src/main/scala/ch/epfl/pop/json/JsonMessageParser.scala

```diff
@@ -51,26 +51,49 @@ object JsonMessageParser {
           obj.getFields("method") match {
             case Seq(JsString(m)) => m match {
               /* Subscribing to a channel */
-              case Methods.Subscribe() => Left(obj.convertTo[SubscribeMessageClient])
+              case Methods.Subscribe() => return Left(obj.convertTo[SubscribeMessageClient])
```

Comment:
Good question! Since we added the ability for the server to parse server messages in e1dc5c1 following the little backend meeting we did, the server should now be able able to parse two types of messages:
- client queries including a `method` field in the message (see line 51 you highlighted)
- answer messages (positive or negative)

The second bullet point corresponds to lines 74-96 of commit e1dc5c1 and is situated *just after* the part you highlighted. Thus, if you parse a query message without the `return`s, then whatever happens, the line will be interpreted as an expression instead of a return value. Finally, it would throw an Exception (l.95).

Did I answer your question @SFulpius ?

Reply:
Yes, I see why you use it. But I'm not so fan of using return in this case as it makes the control flow of the program hard to understand. You could do something like that, which I think is more readable.
```scala
val fields = obj.fields
if (fields.contains("method")) {
    ...
}
else if (fields.contains("error")) {
    ...
}
...
else {
    ...
}
```

Reply:
I hesitated a long time between what I wrote and what you just proposed. The problem I had with the if-else version is that is requires us to check basically twice for "method".

However, looking at it once more, I agree that the `if-else` version is more readable at a glance! I will make the change now